packages/website/blog/2024-04-30-announcing-typescript-eslint-v8-beta.md

```diff
@@ -0,0 +1,226 @@
+
+If you don't yet use typescript-eslint, you can go through our [configuration steps on the v8 _Getting Started_ docs](https://v8--typescript-eslint.netlify.app/getting-started).
+It'll walk you through setting up typescript-eslint in a project.
+
+To use v8 specifically, see the following section for an updated install command.
+
+### As An Existing User
+
+If you already use typescript-eslint, you'll need to first replace your package's previous versions of `@typescript-eslint/eslint-plugin` and `@typescript-eslint/parser` with `@rc-v8` versions:
+
+```shell
+npm i @typescript-eslint/eslint-plugin@rc-v8 @typescript-eslint/parser@rc-v8 --save-dev
+```
+
+We highly recommend then basing your ESLint configuration on the reworked typescript-eslint [recommended configurations mentioned later in this post](#reworked-configuration-names) — especially if it's been a while since you've reworked your linter config.
+
+## User-Facing Changes
+
+These are the changes that users of typescript-eslint -generally, any developer running ESLint on TypeScript code- should pay attention to when upgrading typescript-eslint from v7 to v8.
+
+> ⏳ indicates a change that has been scheduled for v8 but not yet released.
+> We'll update this blog post as the corresponding pull requests land.
+
+### ⏳ ESLint v9 Support
+
+typescript-eslint v8 ships will full support for ESLint v9.
+
+typescript-eslint v7 was our first version that supported ESLint's [new "flat" config file format](https://eslint.org/docs/latest/use/configure/configuration-files), which was already available in ESLint v8.
+ESLint v9 still supports ESLint's [older legacy config file format](https://eslint.org/docs/latest/use/configure/configuration-files-deprecated) so our tooling does as well.
+However, ESLint v9 also includes a set of breaking changes that we added support for in typescript-eslint v8.
+See the [ESLint v9 release blog post](https://eslint.org/blog/2024/04/eslint-v9.0.0-released) and for more details.
+
+### ⏳ Project Service
+
+The biggest new feature added in this version is the stability of our new "project service".
+In short, the project service is a new way to enable [typed linting](/getting-started/typed-linting) that is generally _easier to configure_ and _faster at runtime_ than our previous offerings.
+It's been experimentally available since v6.1.0 under the name `EXPERIMENTAL_useProjectService`; now, we've renamed it to `projectService`.
+
+You can use the new project service in your configuration instead of the previous `parserOptions.project`:
+
+```js title="eslint.config.js"
+import eslint from '@eslint/js';
+import tseslint from 'typescript-eslint';
+
+export default tseslint.config(
+  eslint.configs.recommended,
+  ...tseslint.configs.recommendedTypeChecked,
+  {
+    languageOptions: {
+      parserOptions: {
+        // Remove this line
+        project: true,
+        // Add this line
+        projectService: true,
+        tsconfigRootDir: import.meta.dirname,
```

Comment:
Would `process.cwd()` be a better option here as an example in the world of ESM?

Reply:
No, because the cwd might not match the same directory. Folks might be executing ESLint from somewhere else.